File: scripts/check_divs.js
const fs = require('fs');
const path = 'resources/js/GameBoards/SnapsBoard.tsx';
const text = fs.readFileSync(path, 'utf8');
const lines = text.split(/\r?\n/);
let balance=0;
for (let i=0;i<lines.length;i++){
  const line = lines[i];
  balance += (line.match(/<div/g)||[]).length;
  balance -= (line.match(/<\/div>/g)||[]).length;
  if (balance<0){
    console.log('Negative at', i+1);
    break;
  }
}
console.log('Final balance', balance);

File: scripts/check_divs.py
from pathlib import Path
p=Path('resources/js/GameBoards/SnapsBoard.tsx')
open_count=0
for i,line in enumerate(p.read_text(encoding='utf-8').splitlines(),start=1):
    open_count += line.count('<div')
    open_count -= line.count('</div>')
    if open_count < 0:
        print('Negative at',i)
        break
print('Final balance',open_count)
